contracts/trading/TradingModule.sol

```diff
@@ -246,9 +246,22 @@ contract TradingModule is Initializable, UUPSUpgradeable, ITradingModule {
         decimals = RATE_DECIMALS;
     }
 
+    function _getFlag(uint32 flags, uint16 flagID) private pure returns (bool) {
+        return (flags & flagID) == flagID;
+    }
+
     /// @notice Check if the caller is allowed to execute the provided trade object
-    function canExecuteTrade(Trade calldata trade) external view override returns (bool) {
-        return tokenWhitelist[msg.sender][trade.sellToken].allowSell;
+    function canExecuteTrade(address from, uint16 dexId, Trade calldata trade) external view override returns (bool) {
+        require(dexId <= uint16(DexId.NOTIONAL_VAULT));
```

Comment:
Do you need these two require statements? It's another thing to remember to upgrade if we add another DEX or trade type later.

Reply:
Alternatively, you can execute the same checks in the setTokenPermissions method rather than here. I think that is slightly better from a gas perspective.